WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift

```diff
@@ -439,7 +458,16 @@ private extension OrderDetailsDataSource {
     private func configureShippingLabelProduct(cell: ProductDetailsTableViewCell, at indexPath: IndexPath) {
         cell.selectionStyle = .default
 
-        // TODO-2167: show aggregated order item (product) for a shipping label
+        guard let shippingLabel = shippingLabel(at: indexPath) else {
+            return
+        }
+
+        guard let orderItem = shippingLabelOrderItemsAggregator?.orderItems(of: shippingLabel)[safe: indexPath.row] else {
+            return
+        }
+
+        let itemViewModel = ProductDetailsCellViewModel(aggregateItem: orderItem, currency: order.currency)
+        cell.configure(item: itemViewModel, imageService: imageService)
```

Comment:
The guard statement's failure case is not expected, since it means that there is a miscalculation in the shipping label and/or order item index. I added an `assertionFailure` here: https://github.com/woocommerce/woocommerce-ios/pull/3197/commits/71b2fe4f2d3bce38f6df320a84cd649156abf9c4